### PR TITLE
Request classic jecties

### PR DESCRIPTION
**Pull request by ShiftyRail**
I swear this isn't a meme PR, blame Sonix for giving me the idea if anything

- Can request objectives of the classic type (steal/assasinate/etc)
- They are not checked on round end
- Instead, you get a "jectie" (the food item) depending on if you failed it or not

Tested and most of the code is similar to the other PR, which maybe can be added to the framework later

:cl:
- rscadd: Can request classic objectives ; they are not checked, but you can see if you failed them or not by getting a reward at the end.

### CODE REVIEW

**Comment by SonixApache** [comment]
Won't the name be too long for certain objectives? Might be better to add it to the descriptions.
Did you test it?

**Comment by ShiftyRail** [comment]
Fair point, the objectives I got in solo were all steal objectives, with a relatively short name

**Comment by DeityLink** [comment]
I'll rework this to work with the syndie drone later

**Comment by ShiftyRail** [comment]
Should this PR not go through, I can make a third one with just the framework for this but not have it actively enabled.
I also expended the changelog a bit.

**Comment by ArchSunder** [comment]
we'll have people posting screenshots of their jectie item instead of their greentext

😔

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -1741,6 +1742,28 @@
...
+
+/obj/item/weapon/reagent_containers/food/snacks/jectie/green/New()
+	..()
+	reagents.add_reagent(GREENTEA, 18)
+	reagents.add_reagent(NUTRIMENT, 6)
+	bitesize = 4
+
+// Always red text
+/obj/item/weapon/reagent_containers/food/snacks/jectie/red
+	icon_state = "jectie_green"
```
Comment (kilozombie) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Faulty copypaste?

**Comment by Durgen** [comment]
why wouldn't they be checked?

**Comment by ShiftyRail** [comment]
I feel this would completely negate the point of the original PR.

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: Yep, thank you

**Comment by ShiftyRail** [comment]
Fixed the problem kilozombie mentionned.
Also made classic objective giving a little less baid-aidy and more modular, so that if someone wants to work with this system to add better objectives, or even make the two systems coexist together, it's possible.

**Comment by Tehsapper** [comment]
>this isn't a meme PR
>Instead, you get a "jectie" (the food item) depending on if you failed it or not

?
the original PR was bad enough.

**Comment by ShiftyRail** [comment]
I don't see much difference between getting a green line at the end of the scoreboard and a food item, both exist to tell if you did your objective or not.

**Comment by Lokiamis** [comment]
if there's no difference why don't we just go with the option that people are requesting and isn't just you being snarky
or what if you just didn't act like an obtuse dickhead when people earnestly ask you things

**Comment by Tehsapper** [comment]
you do, others don't.
imagine doing your job as a syndicate operative and all you get is a meme food item.
the immersion is ruined.
the best way to tackle the original issue was to let people have freeform objectives in their character preferences, and not hamfist it to everyone else.

**Comment by ShiftyRail** [comment]
As I said earlier, I feel this would completely negate the point of the original PR.
I'm perfectly fine with someone else doing it, in fact, this basically sets it up for them.

>the immersion is ruined

End of round is out of character already, though.

**Comment by Durgen** [comment]
I don’t see the point of asking for classic jecties if you can’t greentext

**Comment by Kammerjunk** [comment]
before "boring antags who just complete their objectives and then do nothing for the rest of the round" can be used as an argument on the merit of objective state-checking (or objectives in general), you have to understand why said boring antags do it. I don't think it's reasonable to say that the same people who valued their own safety above having fun in a round will change considerably because the "green line at the end" is gone. in fact, I don't think they'll change at all. I think that this assumption unfairly conflates "people who like to get greentext" with "people who will stop playing a round altogether just to eventually get greentext." therefore, I don't think that state-checking those objectives on roundend will negate the point of the original PR

the food is stupid though, and you should think that it's stupid as well because you "don't see much difference between getting a green line at the end of the scoreboard and a food item"

**Comment by Kurfursten** [comment]
I actually think this is fine.

**Comment by jknpj** [comment]
Why even remove jecties if you're just end re-adding them in a really weird way?

**Comment by ShiftyRail** [comment]
>Why even remove jecties if you're just end re-adding them in a really weird way?

People seem very attached to it.

**Comment by DeityLink** [comment]
No shit?

I think objectives being checked at the end however was a mistake in their conception, hence the syndicate drone thingy. Here's how I'm planning to have it work:

1. You use your uplink to request an objective, can be random, or from a category (steal, assassinate, harm, subvert), along as an either random or set difficulty level (harm an assistant / steal a stamp -> kill the HoS / steal a deadly virus sample) **You can only have one active requested objective at once.**
2. Once your deed performed, you use your uplink to call the drone and rendez vous with it to deliver the stolen item / proof of the deed (in the case of harm / kill, it could either be a severed limb or simply a picture of their dead body if you don't want to remove them from the round)
3. The drone then leaves the station, and once it exits the Z level, your objective is marked as successful, and your uplink allows you to pick another objective. No specific rewards.
4. Repeat 1->3 as many times as you wish.

**Comment by ShiftyRail** [comment]
For me, the less structured the role/mode is, the better. 
I fear such a change would make traitor a bit too formulaic.

However we live in a coding democracy after all

**Comment by DeityLink** [comment]
It's a feature that'd you'd be free to ignore, and unlike before, if you ignore them the scoreboard won't display any pointless objectives.

**Pull request closed without merging**